drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -20,8 +21,51 @@
 
 LOG_MODULE_REGISTER(siwx91x_wifi);
 
+#define SIWX91X_INTERFACE_MASK (0x03)
+
 NET_BUF_POOL_FIXED_DEFINE(siwx91x_tx_pool, 1, _NET_ETH_MAX_FRAME_SIZE, 0, NULL);
 
+static inline int siwx91x_get_mode(uint8_t z_mode, sl_wifi_interface_t iface, uint16_t *sl_opermode)
+{
+	switch (z_mode) {
+	case WIFI_STA_MODE:
+		if (FIELD_GET(SIWX91X_INTERFACE_MASK, iface) == SL_WIFI_CLIENT_INTERFACE) {
+			return -EALREADY;
+		}
```

Comment:
I feel this function is bit misleading. Reader does not expect `get_mode()` does such test. Probably you want one function that return `SL_SI91X_CLIENT_MODE` / `SL_SI91X_ACCESS_POINT_MODE` from z_mode and another that return `SL_SI91X_CLIENT_MODE` / `SL_SI91X_ACCESS_POINT_MODE` from the `sl_wifi_interface_t`. Then you can check if `nwp_reboot()` is required or not.